spec/patcher_spec.rb

```diff
@@ -67,21 +85,59 @@ def get_patcher(filename, on_error: :log)
       expect { patcher.interpreter = 'a' }.to raise_error PatchELF::MissingSegmentError
     end
 
-    it 'still executable after patching' do
-      linux_only!
+    shared_examples 'still executable after patching' do |saver_args = {}|
+      it 'still executable after patching' do
+        linux_only!
 
-      %w[pie.elf nopie.elf].each do |f|
-        [0x100, 0xfff].each do |pad_len|
-          patcher = get_patcher(f)
-          patcher.interpreter = (patcher.interpreter + "\x00").ljust(pad_len, 'A')
-          with_tempfile do |tmp|
-            patcher.save(tmp)
-            expect(`#{tmp} < /dev/null`).to eq "It works!\n"
-            expect($CHILD_STATUS.exitstatus).to eq 217
+        %w[pie.elf nopie.elf].each do |f|
+          [0x100, 0xfff].each do |pad_len|
+            patcher = get_patcher(f)
+            patcher.interpreter = (patcher.interpreter + "\x00").ljust(pad_len, 'A')
+            with_tempfile do |tmp|
+              patcher.save(tmp, **saver_args)
+              expect(`#{tmp} < /dev/null`).to eq "It works!\n"
+              expect($CHILD_STATUS.exitstatus).to eq 217
+            end
           end
         end
       end
     end
+
+    context('patchelf_compatible: false') do
+      it_behaves_like 'still executable after patching'
+
+      it 'patches fine but segfaults on execution' do
+        linux_only!
+
+        patcher = get_patcher('syncthing')
+        # TODO: don't run this test on other arch.
+        patcher.interpreter = "/lib64/ld-linux-x86-64.so.2\x00"
```

Comment:
idk what's the standard path for interpreter that covers all architecture.
will this be okay ?

Reply:
There's no reliable universal interpreter path, even on the same architecture (different distros have been known to place it in different locations). I think the unit test should check for the presence of that file and error if it isn't there, since that indicates a test environment issue.

Reply:
> There's no reliable universal interpreter path, even on the same architecture (different distros have been known to place it in different locations).
:sweat: 

>  I think the unit test should check for the presence of that file and error if it isn't there, since that indicates a test environment issue.
agree, guessing by the other ELFs that share the same interpreter path, the tests should run fine on travis.

Reply:
`/lib64/ld-linux-x86-64.so.2` is compatible with all common distributions on x86-64, but there's always weird exceptions.

Reply:
As `syncthing` is a x86-64 binary, I think use `/lib64/ld-linux-x86-64.so.2` is good enough. We only need to ensure it's runnable on CI.